src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -209,58 +173,18 @@ def _concat_db_server_endpoint(db_prefix):
     return _concat_db_server_endpoint
 
 
-def _get_app_insights_telemetry_channel_resource_id(cloud_name):
-    app_insights_telemetry_channel_resource_id_mapper = {
-        'AzureCloud': 'https://dc.applicationinsights.azure.com/v2/track',
-        'AzureChinaCloud': 'https://dc.applicationinsights.azure.cn/v2/track',
-        'AzureUSGovernment': 'https://dc.applicationinsights.us/v2/track'
-    }
-    return app_insights_telemetry_channel_resource_id_mapper.get(cloud_name, None)
-
-
-def _get_log_analytics_resource_id(cloud_name):
-    log_analytics_resource_id_mapper = {
-        'AzureCloud': 'https://api.loganalytics.io',
-        'AzureChinaCloud': 'https://api.loganalytics.azure.cn',
-        'AzureUSGovernment': 'https://api.loganalytics.us'
-    }
-    return log_analytics_resource_id_mapper.get(cloud_name, None)
-
-
-def _get_app_insights_resource_id(cloud_name):
-    app_insights_resource_id_mapper = {
-        'AzureCloud': 'https://api.applicationinsights.io',
-        'AzureChinaCloud': 'https://api.applicationinsights.azure.cn',
-        'AzureUSGovernment': 'https://api.applicationinsights.us'
-    }
-    return app_insights_resource_id_mapper.get(cloud_name, None)
-
+def _get_endpoint_fallback_value(cloud_name):
+    def _get_cloud_endpoint_fallback_value(endpoint_name):
+        endpoint_mapper = {c.name: c.endpoints.__dict__.get(endpoint_name, None) for c in HARD_CODED_CLOUD_LIST}
+        return endpoint_mapper.get(cloud_name, None)
+    return _get_cloud_endpoint_fallback_value
```

Comment:
Is this nested function truly necessary? Can't it be simplified as `_get_endpoint_fallback_value(cloud_name, endpoint_name)`?

`_get_endpoint_fallback_value` does no more than saving `cloud_name` as `_get_cloud_endpoint_fallback_value`'s `locals()`.

See https://stackoverflow.com/questions/4020419/why-arent-python-nested-functions-called-closures